apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx

```diff
@@ -0,0 +1,45 @@
+import { LockedDeviceError } from '@ledgerhq/errors';
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+
+import { useNextMenuUrl } from '../menu/hooks';
+import { ConnectLedgerModal } from './ConnectLedgerModal';
+import {
+    LedgerConnectionFailedError,
+    LedgerNoTransportMechanismError,
+} from './LedgerExceptions';
+
+export function ConnectLedgerModalContainer() {
+    const navigate = useNavigate();
+    const accountsUrl = useNextMenuUrl(true, '/accounts');
+    const importLedgerAccountsUrl = useNextMenuUrl(
+        true,
+        '/import-ledger-accounts'
+    );
+
+    return (
+        <ConnectLedgerModal
+            onClose={() => {
+                navigate(accountsUrl);
+            }}
+            onError={(error) => {
+                navigate(accountsUrl);
+                toast.error(getLedgerErrorMessage(error));
+            }}
+            onConfirm={() => {
+                navigate(importLedgerAccountsUrl);
+            }}
+        />
+    );
+}
+
+function getLedgerErrorMessage(error: unknown) {
+    if (error instanceof LockedDeviceError) {
+        return 'Your device is locked. Un-lock it and try again.';
```

Comment:
`Un-lock` -> `Unlock` 😛

Reply:
thanks haha, this was copy-pasted over and I've been meaning to consolidate all of the error handling/messaging 😛